src/app/cart/cart.component.ts

```diff
@@ -0,0 +1,85 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { NgbRatingConfig } from '@ng-bootstrap/ng-bootstrap';
+import { Cart } from './cart';
+import { CartService } from 'src/app/shared/cart.service';
+import { Observable } from 'rxjs';
+
+@Component({
+  selector: 'app-cart',
+  templateUrl: './cart.component.html',
+  styles: [
+  ]
+})
+export class CartComponent implements OnInit {
+
+  public cartComponentRef = CartComponent;
+
+  @Input() static cartItems : Cart[]= [];
+  
+  totalPrice: number=0;
+
+  constructor(config: NgbRatingConfig, private cart: CartService) { 
+    config.readonly = true;
+    config.max=5;
+  }
+
+  quantityArray: number[] = [1,2,3,4,5];
+
+  async ngOnInit(): Promise<void> {
```

Comment:
Is async and await required for this functionality? Can it not be done without that?

Reply:
changes done